parachain/pallets/basic-channel/src/outbound/mod.rs

```diff
@@ -1,26 +1,85 @@
-use frame_support::{decl_error, decl_event, decl_module, decl_storage,
+use codec::{Encode, Decode};
+use ethabi::{self, Token};
+use frame_support::{
+	decl_error, decl_event, decl_module, decl_storage,
+	weights::Weight,
 	dispatch::DispatchResult,
+	traits::{Get, EnsureOrigin},
+	ensure,
 };
 use frame_system::{self as system};
-use sp_core::H160;
-use sp_std::prelude::*;
-use artemis_core::{
-	ChannelId, MessageNonce, MessageCommitment,
+use sp_core::{H160, H256, RuntimeDebug};
+use sp_io::offchain_index;
+use sp_runtime::{
+	traits::{Hash, Zero, StaticLookup},
 };
+use sp_std::prelude::*;
+
+use artemis_core::{ChannelId, MessageNonce, types::AuxiliaryDigestItem};
+
+#[cfg(feature = "runtime-benchmarks")]
+mod benchmarking;
 
 #[cfg(test)]
 mod test;
 
+/// Wire-format for committed messages
+#[derive(Encode, Decode, Clone, PartialEq, RuntimeDebug)]
+pub struct Message {
+	/// Target application on the Ethereum side.
+	target: H160,
+	/// A nonce for replay protection and ordering.
+	nonce: u64,
+	/// Payload for target application.
+	payload: Vec<u8>,
+}
+
+/// Weight functions needed for this pallet.
+pub trait WeightInfo {
+	fn on_initialize(num_messages: u32, avg_payload_bytes: u32) -> Weight;
+	fn on_initialize_non_interval() -> Weight;
+	fn on_initialize_no_messages() -> Weight;
+	fn set_principal() -> Weight;
+}
+
+impl WeightInfo for () {
+	fn on_initialize(_: u32, _: u32) -> Weight { 0 }
+	fn on_initialize_non_interval() -> Weight { 0 }
+	fn on_initialize_no_messages() -> Weight { 0 }
+	fn set_principal() -> Weight { 0 }
+}
```

Comment:
The "default" implementation doesn't really work like that. Its not a last-resort default-case in case an extrinsic isn't benchmarked.

Each runtime explicitly configures the `WeightInfo` implementations it wants. Here, the default implementation is necessary to configure a working mock runtime for unit tests, which don't really require accurate weight info.